File: Java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven()
    {
        return 40;
    }
    public int remainingMinutesInOven (int timeCooked)
    {
        return 40-timeCooked;
    }
    public int preparationTimeInMinutes(int layers)
    {
        return 2*layers;
    }
    public int totalTimeInMinutes(int layers, int timeCooked)
    {
        return (timeCooked)+(2*layers);
    }
}

File: C-Sharp/Lasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven()
    {
        return 40;
    }
    public int RemainingMinutesInOven(int currentBakeTime)
    {
        return (40-currentBakeTime);
    }
    public int PreparationTimeInMinutes(int layers)
    {
        return (2*layers);
    }
    public int ElapsedTimeInMinutes(int layers, int currentBakeTime)
    {
        return((2*layers)+currentBakeTime);
    }
}
